### PR TITLE
Add admin for FPL namespace to authenticated users - POC

### PR DESCRIPTION
**Pull request by akomisarek**
### Change description ###
As in title. 

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: k8s/aat/common/family-public-law/rolebinding.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+metadata:
+  name: admin-fpl
+  namespace: family-public-law
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: admin
+subjects:
+  - kind: Group
+    name: system:authenticated
```
Comment (timja) [inline k8s/aat/common/family-public-law/rolebinding.yaml:12]: this shouldn't be authenticated it should be the user group

**Comment by timja** [comment]
Has this been approved anywhere? (I'm just looking at it from a code perspective)

**Comment by akomisarek** [comment]
> Has this been approved anywhere? (I'm just looking at it from a code perspective)

Not yet, I am exploring options and testing if that even works - want to open DevOps ticket with problems and investigations we did. Part of it was checking if dev team is empowered to do changes on that level :), which seems not as it is still blocked.

File: k8s/aat/common/family-public-law/rolebinding.yaml (hunk 1 above)
Comment (akomisarek) [reply k8s/aat/common/family-public-law/rolebinding.yaml:12]: Yes, but i guess it needs to be reference by objectId?

**Comment by akomisarek** [comment]
> this doesn't seem to scope it to fpl at all?

I updated the file, not sure if you seen latest version - it binds the role only to FPL namespace.

File: k8s/aat/common/family-public-law/rolebinding.yaml (hunk 1 above)
Comment (akomisarek) [reply k8s/aat/common/family-public-law/rolebinding.yaml:12]: Yes I know, do we have way of dynamically extracting it?

**Comment by akomisarek** [comment]
Ideally I would like to enable extra functionality on namespaces where the teams have got a bit more ownership: i.e. exec on the pods. I was thinking about ClusterRole:
`apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
metadata:
  name: exec-pod
rules:
- apiGroups: [""]
  resources: ["pods/exec"]
  verbs: ["create"]`

And then binding to that role on this level.

**Comment by timja** [comment]
not sure how much benefit exec will give your normally as you're running an image without a shell in most cases, port-forward would be more useful and delete pod / hr.

**Comment by akomisarek** [comment]
Thanks Tim for input, yeah in this instance it would be helpful for threaddump, it seems there is jattach on those images so would be good to use it, but yeah maybe generally port-forward is better, we could connect to JMX even 🤔

**Comment by timja** [comment]
> Thanks Tim for input, yeah in this instance it would be helpful for threaddump, it seems there is jattach on those images so would be good to use it, but yeah maybe generally port-forward is better, we could connect to JMX even 🤔

ah forgot about that, it makes sense to keep exec then

File: k8s/aat/common/family-public-law/rolebinding.yaml (hunk 1 above)
Comment (timja) [reply k8s/aat/common/family-public-law/rolebinding.yaml:12]: we look it up in the pipeline and put it in, you would likely need to hardcode it here

**Pull request closed without merging**